src/InventoryManager.cpp

```diff
@@ -26,203 +26,190 @@ int InventoryManager::userInput() {
-                    break;
-                }
-            } catch (std::exception& e) {
-                /* Catch exception, print out its message, but continue to run as normal. */
-                Logger::logError(e.what());
-                Logger::logWarn("Item %s has not been added to the inventory. Please correct the input and try again.",
-                                name.c_str());
-                break;
-            }
+        if (current_user->permission < 3) {
+            Logger::logWarn("User %s does not have the required permissions to add an Item.",
+                            current_user->name.c_str());
+            return -1;
+        }
 
-            if (active_inventory->addItem(new_item) != -1) {
-                std::cout << "Added " << name << " of type " << category << std::endl;
-                Logger::logTrace("User %s added Item '%s'.", current_user->name.c_str(), name.c_str());
+        std::cout << "Enter item name: ";
+        std::cin >> name;
+        std::cout << "Enter item category (Perishable or NonPerishable): ";
+        std::cin >> category;
+        std::cout << "Enter item sub-category: ";
+        std::cin >> sub_category;
+        std::cout << "Enter item quantity: ";
+        std::cin >> quantity;
+        std::cout << "Enter backorder (set to zero unless there is negative stock): ";
+        std::cin >> backorder;
+        std::cout << "Enter item id: ";
+        std::cin >> id;
+        std::cout << "Enter sale price: (format xx.xx) $";
+        std::cin >> sale_price;
+        std::cout << "Enter purchase cost: (format xx.xx) $";
+        std::cin >> buy_price;
+        std::cout << "Enter item tax as a decimal value: ";
+        std::cin >> tax;
+        std::cout << "Enter expiration date (format xx/xx/xxxx) or -1 for NonPerishable: ";
+        std::cin >> expiration;
+
+        lowerCaseString(category);
+        try {
+            if (category == "perishable") {
+                new_item = std::make_shared<PerishableItem>(name, "Perishable", sub_category, quantity, backorder,
+                                                            id, sale_price, buy_price, tax, expiration);
+            } else if (category == "nonperishable") {
+                new_item = std::make_shared<NonPerishableItem>(name, "NonPerishable", sub_category, quantity,
+                                                                backorder, id, sale_price, buy_price, tax);
+            } else {
+                throw std::runtime_error("Invalid category.");
+                return -1;
             }
-            break;
+        } catch (std::exception& e) {
+            /* Catch exception, print out its message, but continue to run as normal. */
+            Logger::logError(e.what());
+            Logger::logWarn("Item %s has not been added to the inventory. Please correct the input and try again.",
+                            name.c_str());
+            return -1;
```

Comment:
So, returns are correct -- but we need to the change return value given the way the program is currently structured.

Currently, returning -1 exits the program.  We don't want to exit on most errors.  The ```break``` statements you replaced with ```return -1```, try replacing with ```return 0```

Reply:
ok i went back n fixed it!